modules/aws/src/smithy4s/query/AwsSchemaVisitorAwsQueryCodec.scala

```diff
@@ -0,0 +1,161 @@
+package smithy4s.aws.query
+
+import smithy4s.schema._
+import smithy.api.{XmlFlattened, XmlName}
+import smithy4s.{Schema => _, _}
+
+private[aws] class AwsSchemaVisitorAwsQueryCodec(
+    val cache: CompilationCache[AwsQueryCodec]
+) extends SchemaVisitor.Cached[AwsQueryCodec] { compile =>
+
+  override def primitive[P](
+      shapeId: ShapeId,
+      hints: Hints,
+      tag: Primitive[P]
+  ): AwsQueryCodec[P] =
+    (p: P) => {
+      Primitive.stringWriter(tag, hints) match {
+        case Some(writer) => FormData.simple(writer(p))
+        case None         => FormData.empty
+      }
+    }
+
+  override def collection[C[_], A](
+      shapeId: ShapeId,
+      hints: Hints,
+      tag: CollectionTag[C],
+      member: Schema[A]
+  ): AwsQueryCodec[C[A]] = {
+    val memberWriter = compile(member)
+    val key = if (hints.has[XmlFlattened]) {
+      Nil
+    } else {
+      List(getKey(hints, "member"))
+    }
+    println(s"KEY: ${key}")
+
+    (collection: C[A]) =>
+      tag
+        .iterator(collection)
+        .zipWithIndex
+        .map { case (member, index) =>
+          val newKey = ((index + 1).toString :: key).reverse.mkString(".")
```

Comment:
here you should just be doing : 

``` 
memberWriter(member).prepend(index) 
```

there's no need to reverse, nor to call `.mkString` prematurely. The `makeString` call will happen naturally when the `.render` is called on `FormData`

Reply:
Done